### DISCUSSION TITLE
Does multimap support device_ref?

### DISCUSSION THREAD

**Discussion by Po1sson** [NVIDIA/cuCollections#545] [Q&A] [answered]
I don't know if this question is stupid, because I'm a beginner and can't really understand the code.
My question is, does multimap support the device_ref function? I want to perform an insert operation in a custom kernel function, but I found that map, set, and multiset all have corresponding map.ref that can be used, but multimap doesn't seem to have one.
How can I insert a key-value pair into a custom type multimap in my own kernel function?

**Comment by PointKernel** [comment, marked as answer]
> How can I insert a key-value pair into a custom type multimap in my own kernel function?

Please check out this example: https://godbolt.org/z/vY3fq8GP8

We are deprecating the legacy data structures and will soon refactor `static_multimap` with the new ref design.

**Reply by Po1sson** [reply]
Thanks so much for your help, and forgive me for the stupid question, I just found out from the comments that I can use device_view to achieve my purpose.
This is a great project and many people will benefit from it.
